include/CXXGraph/Graph/Graph_impl.hpp

```diff
@@ -232,22 +232,23 @@ void Graph<T>::removeEdge(const CXXGraph::id_t edgeId) {
 template <typename T>
 void Graph<T>::removeNode(const std::string &nodeUserId) {
   auto nodeOpt = getNode(nodeUserId);
-  auto isolatedNodeIt = isolatedNodesSet.find(nodeOpt.value());
-
-  if (nodeOpt.has_value() && isolatedNodeIt != isolatedNodesSet.end()) {
-    // The node is isolated
-    isolatedNodesSet.erase(isolatedNodeIt);
-  } else if (nodeOpt.has_value()) {
-    // The node is not isolated
-    // Remove the edges containing the node
-    auto edgeset = edgeSet;
-    for (auto edgeIt : edgeset) {
-      if (edgeIt->getNodePair().first->getUserId() == nodeUserId ||
-          edgeIt->getNodePair().second->getUserId() == nodeUserId) {
-        this->removeEdge(edgeIt->getId());
+  if (nodeOpt) {
+    auto isolatedNodeIt  = isolatedNodesSet.find(nodeOpt.value());
+    if (isolatedNodeIt != isolatedNodesSet.end()) {
+      // The node is isolated
+      isolatedNodesSet.erase(isolatedNodeIt);
+    } else {
+      // The node is not isolated
+      // Remove the edges containing the node
+      auto edgeset = edgeSet;
+      for (auto edgeIt : edgeset) {
```

Comment:
Here I would use `const auto &` instead of `auto`

Reply:
Hi, yes I understand that - no need to make a copy of the edge set, accidentally missed it. When I make that change and use const auto & edgeset = edgeSet, I run into a segmentation fault on the Test_connectedNode test. Not sure how to deal with it. Trying to look through the code to see what could be up

Reply:
This is likely due to the iterator being invalidated after a successful removal (https://stackoverflow.com/a/16904496/1078527).

The solution is to increment the iterator (advance it by one), such that it no longer points to the invalidated element before the next iteration.

Reply:
Note, that you'll likely need to use the explicit `for` loop, instead of iterator-enhanced:
```
for (auto edgeIt = edgeSet.begin(); edgeIt != edgeSet.end(); ++edgeIt) { ...
```

This is necessary because `std::unordered_set` automatically dereferences iterators in iterator-enhanced for loops (which is why it's crashing).

This also means you'll need a dereference on every iterator access, i.e. `(*edgeIt)->`

Reply:
See https://stackoverflow.com/questions/8234779/how-to-remove-from-a-map-while-iterating-it